static/app/views/dashboards/create.tsx

```diff
@@ -42,7 +42,8 @@ function CreateDashboard(props: Props) {
     if (constructedWidget) {
       browserHistory.replace(location.pathname);
     }
-  }, [location.pathname]);
+  }, [location.pathname, location.query]);
```

Comment:
The missing dep here was causing a lint error.

Reply:
I'll double check that.

Reply:
I did some manual testing and created new dashboards, added widgets to them and cloned dashboards and didn't encounter any redirection loops.